### PR TITLE
feat(site): Ask for parameter values when update a workspace

### LINKED ISSUE #6100: Rich parameters: mark parameter required
The goal of this issue is to enable optional rich parameters. At the moment all parameters are required and there is no easy way to pass an empty value.

Most likely we will need to add an extra property `required` to terraform-provider-coder.

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Demo:

https://user-images.githubusercontent.com/3165839/224786903-cfa71349-814c-4cd5-9814-74bc54ee4523.mov

Closes #6100

### CODE REVIEW

File: site/src/pages/WorkspacePage/UpdateBuildParametersDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+      <DialogTitle
+        id="update-build-parameters-title"
+        classes={{ root: styles.title }}
+      >
+        Workspace parameters
+      </DialogTitle>
+      <DialogContent className={styles.content}>
+        <DialogContentText className={styles.info}>
+          It looks like the new version has some mandatory parameters that need
+          to be filled in to update the workspace.
```
Comment (mtojek) [inline site/src/pages/WorkspacePage/UpdateBuildParametersDialog.tsx:67]: nit: translation?

File: site/src/util/richParameters.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    const buildParameter: WorkspaceBuildParameter = {
+      name: parameter.name,
+      value: parameterValue || "",
+    }
+    defaults.push(buildParameter)
+  })
+  return defaults
+}
+
+export const ValidationSchemaForRichParameters = (
```
Comment (mtojek) [inline site/src/util/richParameters.ts:46]: nit: Does it still need to be upper-case?

File: site/src/testHelpers/handlers.ts [hunk 3]
```diff
@@ -311,4 +311,11 @@ export const handlers = [
   rest.get("/api/v2/deployment/stats", (_, res, ctx) => {
     return res(ctx.status(200), ctx.json(M.MockDeploymentStats))
   }),
+
+  rest.get(
+    "/api/v2/workspacebuilds/:workspaceBuildId/parameters",
+    (_, res, ctx) => {
+      return res(ctx.status(200), ctx.json([M.MockTemplateVersionParameter1]))
```
Comment (mtojek) [inline site/src/testHelpers/handlers.ts:318]: Something is fishy here...

`/api/v2/workspacebuilds/:workspaceBuildId/parameters` should return an array of `MockWorkspaceBuildParameter...`.

**Review by mtojek — APPROVED** [review]
Reviewing your PRs always brings educational value to you. Thanks! 

I left one nitpick and raised one question, but otherwise it's 👍 .

**Comment by mtojek** [comment]
Oh, CI complained about:

```
FAIL test src/pages/WorkspacePage/WorkspacePage.test.tsx (41.707 s)
  ● WorkspacePage › after an update postWorkspaceBuild is called with the latest template active version id

    expect(jest.fn()).toBeCalledWith(...expected)

    Expected: "test-outdated-workspace", "test-template-version"

    Number of calls: 0

    Ignored nodes: comments, script, style
```

Run: https://github.com/coder/coder/actions/runs/4408015320/jobs/7722465954

File: site/src/testHelpers/handlers.ts (hunk 3 above)
Comment (BrunoQuaresma) [reply site/src/testHelpers/handlers.ts:318]: Ahhhh, ok! Thank you.

**Pull request merged**